## **0. Introduction**

This notebook serves the purpose of exploring the data used in the master thesis. For now I will look into the dataset from [Kosti et al., 2017](https://openaccess.thecvf.com/content_cvpr_2017/papers/Kosti_Emotion_Recognition_in_CVPR_2017_paper.pdf), the data can be requested [here](https://docs.google.com/forms/d/e/1FAIpQLScXwxhEZu7RpHwgiRqVfb09GzHSSyIm64hJQMgHSLm75ltsFQ/viewform).

There are different types of image folders. At first I will investigate the "emodb_small" folder. This contains mostly images, in which the face is clearly visible.

In [12]:
import os
os.chdir('/mnt/c/Users/janos/OneDrive/Desktop/Master_thesis/kosti/emodb_small/images')

In [14]:
len(os.listdir())

861

The emod_small folder contains 861 images.

**Note:**

The annotations are stored in a *struct* within a *.mat* file. There exists a pytorch implementation, that stores .py file which transform the mat files into csv files. Please follow [this link](https://github.com/Tandon-A/emotic) to the respective github repository. After I assembled the images and annotations in the demanded structure, I ran the mat2py file and ended up with a training, validation and test csv file.

In [15]:
os.chdir('/mnt/c/Users/janos/OneDrive/Desktop/Master_thesis/kosti/mat2csv/emotic_pre/')

In [16]:
os.listdir()

['test.csv', 'train.csv', 'val.csv']

We will now inspect the 'train.csv' file.

In [17]:
import pandas as pd
train = pd.read_csv('train.csv')

In [29]:
print('The training data consists of {} images.'.format(train.shape[0]))

The training data consists of 23266 images.


In [26]:
train.head(n=10)

Index         Folder                         Filename  Image Size  \
0      0  mscoco/images    COCO_val2014_000000562243.jpg  [640, 640]   
1      1  mscoco/images  COCO_train2014_000000288841.jpg  [640, 480]   
2      2  mscoco/images    COCO_val2014_000000558171.jpg  [640, 480]   
3      3  mscoco/images  COCO_train2014_000000369575.jpg  [480, 640]   
4      4  mscoco/images  COCO_train2014_000000213009.jpg  [500, 333]   
5      5  mscoco/images  COCO_train2014_000000462955.jpg  [640, 478]   
6      6  mscoco/images    COCO_val2014_000000168683.jpg  [500, 375]   
7      7  mscoco/images  COCO_train2014_000000186198.jpg  [640, 429]   
8      8  mscoco/images  COCO_train2014_000000006590.jpg  [640, 480]   
9      9  mscoco/images  COCO_train2014_000000144608.jpg  [350, 500]   

                   BBox                         Categorical_Labels  \
0    [86, 58, 564, 628]       ['Disconnection', 'Doubt/Confusion']   
1  [485, 149, 605, 473]                           ['Anticipation']   
2   [305, 92, 461, 465]  ['Engagement', 'Excitement', 'Happiness']   
3   [221, 63, 448, 372]                   ['Aversion', 'Pleasure']   
4   [44, 143, 150, 288]               ['Confidence', 'Excitement']   
5    [42, 32, 413, 472]    ['Anticipation', 'Engagement', 'Peace']   
6   [257, 39, 405, 183]             ['Anticipation', 'Engagement']   
7   [336, 80, 494, 327]                  ['Anticipation', 'Peace']   
8  [188, 109, 381, 382]                             ['Engagement']   
9   [198, 29, 300, 232]                   ['Fatigue', 'Happiness']   

  Continuous_Labels  Gender       Age  
0         [5, 3, 9]    Male     Adult  
1         [6, 4, 7]    Male     Adult  
2         [7, 8, 8]    Male  Teenager  
3         [8, 9, 8]    Male       Kid  
4        [7, 9, 10]    Male     Adult  
5         [3, 6, 8]    Male     Adult  
6         [6, 7, 7]    Male     Adult  
7         [7, 7, 8]    Male       Kid  
8         [7, 4, 7]  Female       Kid  
9         [7, 7, 6]    Male  Teenager

We see that the training set not only contains the continous and categorial labels, but the respective image. Since we may want to take adavantage of the different image type for each "folder", we could extract the respective folder with the labels. Meaning we have 4 csv files for each folder type. 

In [24]:
train['Folder'].unique()

array(['mscoco/images', 'emodb_small/images', 'framesdb/images',
       'ade20k/images'], dtype=object)